Userland/Libraries/LibWeb/HTML/POSTResource.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2022, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteBuffer.h>
+
+namespace Web::HTML {
+
+// https://html.spec.whatwg.org/multipage/browsing-the-web.html#post-resource
+struct POSTResource {
+    // https://html.spec.whatwg.org/multipage/browsing-the-web.html#post-resource-request-body
+    // A request body, a byte sequence or failure.
+    Optional<ByteBuffer> request_body;
```

Comment:
This doesn't seem to be reflecting the notion of `a byte sequence or failure`. From the spec this seems supposed to be a function returning `ErrorOr<Optional<ByteBuffer>>`. If that might be something that is pushed on the yak stack for the future, it's maybe worth a note or a fixme.

Reply:
right, I changed it to `Variant<Empty, RequestBodyFailureTag, ByteBuffer>`. according to how this property is used in the spec we don't need `ErrorOr` because there is no need to know the reason of failure.

thank you for having a look!

Reply:
I think a FIXME would suffice here until we see how this will be implemented, unless you actually use `RequestBodyFailureTag` anywhere. It was this part that made me reason it should be a function, but until we know it's better to leave it as a FIXME:

>This is only ever accessed [in parallel](https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel), so it doesn't need to be stored in memory. However, it must return the same [byte sequence](https://infra.spec.whatwg.org/#byte-sequence) each time. If this isn't possible due to resources changing on disk, or if resources can no longer be accessed, then this must be set to failure.